core/src/banking_stage/unprocessed_transaction_storage.rs

```diff
@@ -263,10 +263,35 @@ impl UnprocessedTransactionStorage {
     }
 
     pub fn new_bundle_storage(
-        unprocessed_bundle_storage: VecDeque<ImmutableDeserializedBundle>,
-        cost_model_failed_bundles: VecDeque<ImmutableDeserializedBundle>,
+        unprocessed_bundle_storage: VecDeque<(
+            u8, /* attempts remaining */
+            ImmutableDeserializedBundle,
+        )>,
+        cost_model_failed_bundles: VecDeque<(
+            u8, /* attempts remaining */
+            ImmutableDeserializedBundle,
+        )>,
+    ) -> Self {
+        Self::new_bundle_storage_ttl(
```

Comment:
nit: why does this need to be a separate function call?

Reply:
this was to avoid modifying all the call sites with the TTL info